xbmc/ServiceManager.cpp

```diff
@@ -155,6 +159,11 @@ bool CServiceManager::InitStageTwo(const CAppParamParser &params, const std::str
 // stage 3 is called after successful initialization of WindowManager
 bool CServiceManager::InitStageThree(const std::shared_ptr<CProfileManager>& profileManager)
 {
+  // Initializes WS-Discovery daemon (only platform Win32)
+#ifdef TARGET_WINDOWS_DESKTOP
+  CWSDiscoverySupport::Get()->Initialize();
```

Comment:
What about moving this to the windows specific init to avoid ifdeffing in common code?

Reply:
I had done it in the beginning but Jenkins doesn't like it. 
It also needs to be started a bit later that Platform->Init() so that it is visible in the logs. Maybe Platform->Init_2()   and PlatformWin32 override ???

Is like this on a provisional basis to rule out Jenkins problems.... He still doesn't like it. I need to squash into a single commit as that is likely the cause.

Reply:
Then move it to the windowing init :)

Reply:
And something like that?

```c++
class CPlatform
{
public:
  /**\brief Creates the Platform object
   *
   *@return the platform object
  */
  static CPlatform *CreateInstance();

  /**\brief C'tor */
  CPlatform();

  /**\brief D'tor */
  virtual ~CPlatform();

  /**\brief Called at an early stage of application startup
   *
   * This method can be used to do platform specific environment preparation
   * or initialisation (like setting environment variables for example)
   */
  virtual bool Init();

  /**\brief Called at a later stage of application startup
  *
  * This method can be used to do platform specific services initialization
  * that depends of others services/resources initialized at Init()
  */
  virtual bool Init_2(){};

  /**\brief Called at an early stage of application shutdown
  *
  * This method can be used to do platform specific services termination
  */
  virtual bool Term_2(){};

  /**\brief Flag whether disabled add-ons - installed via packagemanager or manually - should be
   * offered for configuration and activation on kodi startup for this platform
   */
  virtual bool IsConfigureAddonsAtStartupEnabled() { return false; };
};
```

Reply:
I'd rather not really. The windowing init is basically a second platform init already.

Reply:
> Looks good to me except the handle to the daemon being stored in winsystem

But then it will have to be left where it was at the beginning. And it is true that it has nothing to do with windowing / graphics.

@Paxxi: thoughts?

Reply:
I think the ifdef solution with service broker is the correct one.

Ideally we could abstract away the ifdef into some platform specific place. That's a bit of work as I don't believe we really have a good reusable solution for it.

Leaving it as is is best in my opinion.

Reply:
we spend so much time creating platform agnostic interfaces only to introduce platform ifdefs back in :frowning_face: 

So no, I don't believe `CServiceManager` is the place for this. Any of the other platform specific places is okay with me.

Reply:
And why not 2 Platform initialization stages?  This can be reused/useful for others platforms/cases

Reply:
Stick it in windowing then for now

Reply:
OKAY. Tomorrow I will... and maybe tomorrow the Jenkins issue will be fixed too.

Reply:
Done. Remains to fix the Jenkins issues: Failure after compiling OK on Android / OSX / TVOS

> 14:17:47 Build step 'Execute shell' marked build as failure